src/postamble_minimal.js

```diff
@@ -276,16 +280,33 @@ WebAssembly.instantiate(Module['wasm'], imports).then(/** @suppress {missingProp
 
   initRuntime(wasmExports);
 
-#if PTHREADS && PTHREAD_POOL_SIZE
-  var workersReady = PThread.loadWasmModuleToAllWorkers();
-#if PTHREAD_POOL_DELAY_LOAD
-  ready();
-#else
-  workersReady.then(ready);
-#endif
-#else
-  ready();
+{{{ function waitOnStartupPromisesAndEmitReady() {
+  var promises = [];
+  if (PTHREADS && PTHREAD_POOL_SIZE) {
+    promises.push('PThread.loadWasmModuleToAllWorkers()');
+  }
+  if (LOAD_SOURCE_MAP) {
+    promises.push('getSourceMapAsync().then(json=>{receiveSourceMapJSON(json)})');
+  }
+  if (promises.length == 0) {
+    return 'ready();'
+  } else if (promises.length == 1) {
+    return `${promises[0]}.then(ready);`;
+  } else {
+    return `Promise.all(${', '.join(promises)}).then(ready);`
+  }
+}
+null;
+}}}
```

Comment:
There are now two Promises to possibly wait on, so add a helper to emit them at compile-time.